Update Hacktoberfest related pages for 2023 (#6668)
jenkins-infra/jenkins.io: content/events/hacktoberfest.adoc

  - programs
  - hacktoberfest
links:
  discourse: https://community.jenkins.io/t/hacktoberfest-2023/9613
opengraph:
  image: /images/hacktoberfest/hacktoberfest_2022_social.png
---

image:/images/hacktoberfest/hacktoberfest_2023_logo.png[Hacktoberfest, role=center]

link:https://hacktoberfest.com/[Hacktoberfest]
is a month-long celebration of open source software that happens every October.

> During this event, everyone can support open source by contributing changes and earn a unique digital reward.

Reviewer: ```suggestion
During this event, everyone can support open source by contributing changes and earning a unique digital reward.
```
adjust for proper tense usage

This year marks the 10th anniversary of Hacktoberfest!

We invite everyone to participate and contribute to the Jenkins project, regardless of their background and Jenkins experience.
There are many ways to link:/participate/[contribute] to Jenkins during Hacktoberfest.
You can participate by working on link:https:/participate/code/[code] or link:/participate/document/[documentation], contributing to link:/doc/developer/internationalization/[localization], or creating new link:/artwork[artwork].
You can also share your experiences with Jenkins by link:https://github.com/jenkins-infra/jenkins.io/blob/master/CONTRIBUTING.adoc#adding-a-blog-post[creating a new blogpost].

> Generally, any four non-spammy pull requests in GitHub or merge requests in GitLab would qualify.

Reviewer 2: Since the Jenkins project has no presence on GitLab.com, I think we should consider only mentioning GitHub.
```suggestion
Generally, any four non-spammy pull requests in GitHub would qualify.
```

Please refer to the link:/participate/[Contribute and Participate] page for more information about how to contribute.

== Quick start

1. Sign-up to Hacktoberfest on link:https://hacktoberfest.com[the event website].
2. Join link:https://app.gitter.im/#/room/#jenkinsci_hacktoberfest:gitter.im[our Gitter channel].

> 3. Everything is set, just start creating pull or merge requests!

Reviewer 2: A merge request is a GitLab term.  I'm OK if we use it, though I believe that almost everyone in the reading audience will recognize the term "pull request".
```suggestion
3. Everything is set, just start creating pull requests!
```

** If a repository has no `hacktoberfest` topic set, contact the maintainers to see if they are willing to accept a contribution.

> ** Alternatively, if a pull or merge request has the label `hacktoberfest-accepted`, it will count towards your total.

Reviewer 2: ```suggestion
** Alternatively, if a pull request has the label `hacktoberfest-accepted`, it will count towards your total.
```

> ** Your pull or merge request must not be labeled as invalid to be counted.

Reviewer 2: ```suggestion
** Your pull request must not be labeled as invalid to be counted.
```

> ** For your pull or merge requests to be accepted, the request must either be merged, have the `hacktoberfest-accepted` label, or have an overall approving review on GitHub.

Reviewer 2: ```suggestion
** For your pull requests to be accepted, the request must either be merged, have the `hacktoberfest-accepted` label, or have an overall approving review on GitHub.
```

== Where to contribute?

> The Jenkins project is spread across two organizations on GitHub: `link:https://github.com/jenkinsci/[jenkinsci]` and `link:https://github.com/jenkins-infra/[jenkins-infra]`.

Reviewer 2: I think that we want to make the text of the link as code, not the entire link
```suggestion
The Jenkins project is spread across two organizations on GitHub: link:https://github.com/jenkinsci/[`jenkinsci`] and link:https://github.com/jenkins-infra/[`jenkins-infra`].
```

  image: /images/hacktoberfest/hacktoberfest_2022_social.png
---

image:/images/hacktoberfest/hacktoberfest_2023_logo.png[Hacktoberfest, role=center]

link:https://hacktoberfest.com/[Hacktoberfest]
is a month-long celebration of open source software.
It happens every year in October.
During this event everyone can support open-source by contributing changes, and then earn a unique digital reward.
This year marks the 10th anniversary of Hacktoberfest.

We invite everyone to participate and to contribute to the Jenkins project, regardless of their background and Jenkins experience.
There are many ways to link:/participate/[contribute] to Jenkins during Hacktoberfest.

> You can work on code or documentation, contribute to link:https://wiki.jenkins.io/display/JENKINS/Internationalization[localization], or create new link:/artwork[artwork], etc.

Reviewer 3: ```suggestion
You can work on code or documentation, contribute to link:/doc/developer/internationalization/, or create new link:/artwork[artwork], etc.
```

You can also share your experiences with Jenkins by link:https://github.com/jenkins-infra/jenkins.io/blob/master/CONTRIBUTING.adoc#adding-a-blog-post[creating a new blogpost].
Generally, any four non-spammy pull requests in GitHub or merge requests in GitLab would qualify.
Please see the link:/participate/[Contribute and Participate] page for more information about how to contribute.

== Quick start

1. Sign-up to Hacktoberfest on link:https://hacktoberfest.com[the event website].

> 2. Join link:https://matrix.to/#/#jenkinsci_hacktoberfest:gitter.im[our Gitter channel].

Reviewer 3: Why was the link changed from a ready-to-use service to the matrix URL shortener?

Author: The change was intentional so as to provide a more modern platform / medium for people to connect with us, as we have received complaints about the Gitter links before.

Reviewer 3: > was intentional so as to provide a more modern platform / medium for people to connect with us

![Screenshot 2023-09-13 at 17 02 37](https://github.com/jenkins-infra/jenkins.io/assets/13383509/b9f3935a-7764-4066-ab0f-90106382356e)

I don't see any way to connect with anyone here, that's just a URL shortener when the former already opens a web client with a chat you can start talking in.

>  as we have received complaints about the Gitter links

I'm interested in what the issues are, which would be addressed by replacing the chat window with a URL shortener pointing to a generic destination.

Reviewer 4: @krisstern & @NotMyFault 
I propose to keep the link as it was before. This will give us a little more time to debate the policy with Gitter links as both sides have valuable arguments.
This should be solved for when we start the GSoC-2024 campaign where Gitter is heavily used in the various pages.

WDYT ?

Author: Sure @jmMeessen I don't have a very strong opinion about it either way.

Reviewer 3: > I propose to keep the link as it was before.

Makes sense.

3. Everything is set, just start creating pull or merge requests!
** If a repository has no `hacktoberfest` topic is set, contact the maintainers to see if they are willing to accept a contribution.
** Alternatively, if a pull or merge request has the label `hacktoberfest-accepted`, it will count towards your total.
** Your pull or merge request must not be labeled as invalid to be counted.

> ** For your pull or merge requests to be accepted, your pull or merge requests must be merged, or have the `hacktoberfest-accepted` label, or have an overall approving review on GitHub.

Reviewer: ```suggestion
** For your pull or merge requests to be accepted, the request must either be merged, have the `hacktoberfest-accepted` label, or have an overall approving review on GitHub.
```
adjusting for syntax, removing repetitive language

> ** For your pull or merge requests to be accepted, your pull or merge requests must be merged, or have the `hacktoberfest-accepted` label, or have an overall approving review on either GitHub or GitLab.

Reviewer 3: ```suggestion
** For your pull or merge requests to be accepted, your pull or merge requests must be merged, or have the `hacktoberfest-accepted` label, or have an overall approving review on GitHub.
```
The Jenkins project has no repositories on GitHub.

Reviewer 4: You mean Gitlab, @NotMyFault ? 😉

== Where to contribute?

> The Jenkins project is spread across several organizations on GitHub (`link:https://github.com/jenkinsci/[jenkinsci]`, `link:https://github.com/jenkins-infra/[jenkins-infra]`, `link:https://github.com/jenkins-zh/[jenkins-zh]`).

Reviewer 3: ```suggestion
The Jenkins project is spread across several organizations on GitHub (`link:https://github.com/jenkinsci/[jenkinsci]` and `link:https://github.com/jenkins-infra/[jenkins-infra]`.).
```
Let's not promote something we don't monitor.

Reviewer 4: Agreed to remove all references to the ZH repo as it is untracked.
(nit-picking): the added full-stop (after jenkins-infra) is not correct (at least in english) AFAIK

> The Jenkins project is spread across several organizations on GitHub (`link:https://github.com/jenkinsci/[jenkinsci]` and `link:https://github.com/jenkins-infra/[jenkins-infra]`.).
> You are welcome to contribute to **any** repository in **any** of those organizations, or any other Jenkins-related repository on GitHub.
> Repositories may have different contribution guidelines, review and merge policies.

Reviewer: ```suggestion
The Jenkins project is spread across two organizations on GitHub: `link:https://github.com/jenkinsci/[jenkinsci]` and `link:https://github.com/jenkins-infra/[jenkins-infra]`.
You are welcome to contribute to *any* repository in *either* of those organizations or any other Jenkins-related repository on GitHub.
Be aware that repositories may have different contribution guidelines, review processes, and merge policies.
```

I'm updating the text since there are only two organizations now listed here, and adjusting some of the other text.

> If you adopt Jenkins in your open-source projects (e.g. Jenkins Pipeline or Configuration as Code), it counts as well!
> Note that not all pull requests will automatically count towards Hacktoberfest.

Reviewer: ```suggestion
If you adopt Jenkins in your open-source projects, such as Jenkins Pipeline or Configuration as Code, this counts as well!
Not all pull requests will automatically count towards Hacktoberfest.
Keep in mind, the pull or merge request must either be merged, have the `hacktoberfest-accepted` label, or have an overall approving review on GitHub.
```
suggestion for updating this text a bit with the criteria stated above

> * link:https://github.com/search?q=org%3Ajenkinsci+org%3Ajenkins-infra+org%3Ajenkins-zh+topic%3Ahacktoberfest[List of repositories marked for Hacktoberfest] - you can just submit pull requests, no extra steps needed.

Reviewer 3: ```suggestion
* link:https://github.com/search?q=org%3Ajenkinsci+org%3Ajenkins-infra+topic%3Ahacktoberfest&type=repositories[List of repositories marked for Hacktoberfest] - you can just submit pull requests, no extra steps needed.
```

> * link:https://github.com/search?q=org%3Ajenkinsci+org%3Ajenkins-infra+topic%3Ahacktoberfest&type=repositories[List of repositories marked for Hacktoberfest] - you can just submit pull requests, no extra steps needed.

Reviewer: ```suggestion
* link:https://github.com/search?q=org%3Ajenkinsci+org%3Ajenkins-infra+topic%3Ahacktoberfest&type=repositories[Repositories marked for Hacktoberfest] - In these repositories you can just submit pull requests, no extra steps are needed.
```
adjusting some text for readability

> * For other repositories, we will need to get the repo marked with the topic "Hacktoberfest" for your pull request to count towards the goal.

Reviewer: ```suggestion
* For other repositories, we will need to get the repo marked with the "Hacktoberfest" label for your pull request to count towards the goal.
```
I _think_ it's considered a label, but I could be wrong.

Reviewer 4: For repositories it is a "topic" while PRs and issues get a "label" in Github terminology AFAIK.

Reviewer: Ahh, that makes more sense, that's where my confusion was. thanks very much @jmMeessen !

  Please follow link:/events/hacktoberfest/faq/#how-do-i-mark-my-pull-requests[FAQ: Marking Pull requests].

> // === Issue queries

Reviewer 3: What's the purpose of this section?

Reviewer 4: The purpose is/was to give direct links to issue queries for entry-level issues. 
The issues are currently being reviewed/triaged.

So the list/section should be published (when ready) in a second iteration.

|=========================================================

=== Experienced developers

> If you are an established developer and want to create something new, please don't let yourself to be blocked by the suggested topics!

Reviewer: ```suggestion
If you are an established developer and want to create something new, feel free to explore beyond the suggested topics!
```
suggestion to change the text so that it makes a bit more sense, while still giving the same encouragement

Feel free to contribute to any area of Jenkins.
If you see any major functionality missing in Jenkins, we invite you to create new plugins.

> See the link:/doc/developer/tutorial/[Plugin Tutorial] and link:/doc/developer/publishing/requesting-hosting/[Hosting Plugins] guidelines for more information.

Reviewer: ```suggestion
Refer to the link:/doc/developer/tutorial/[Plugin Tutorial] and link:/doc/developer/publishing/requesting-hosting/[Hosting Plugins] guidelines for more information.
```
Suggestion to update `see` to `refer to` for accessibility/inclusion purposes

== Contact us

> * Gitter: link:https://matrix.to/#/#jenkinsci_hacktoberfest:gitter.im[jenkinsci/hacktoberfest]

Reviewer 3: Same question like above.
